admin-purchase-options-action/src/PurchaseOptionsActionExtension.liquid

```diff
@@ -0,0 +1,297 @@
+{%- if flavor contains "react" -%}
+import {useState} from 'react';
+import {
+  useApi,
+  AdminAction,
+  BlockStack,
+  Button,
+  TextField,
+  ChoiceList,
+  Box,
+  InlineStack,
+  NumberField,
+  Select,
+} from '@shopify/ui-extensions-react/admin';
+
+export default function PurchaseOptionsActionExtension(extension) {
+  // The useApi hook provides access to several useful APIs like i18n, close, and data.
+  const {i18n, close, data} = useApi(extension);
+  console.log({data});
+  const [merchantCode, setMerchantCode] = useState('');
+  const [planName, setPlanName] = useState('');
+  const [discountType, setDiscountType] = useState('percentageOff');
+  const [deliveryOptions, setDeliveryOptions] = useState({
+    frequency: 0,
+    timeType: 'day',
+    discount: 0,
+  });
+
+  const updateDeliveryOption = (field, value) => {
+    setDeliveryOptions((prevOptions) => ({
+      ...prevOptions,
+      [field]: value,
+    }));
+  };
+
+  function handleSave() {
+    // This is where you can use the sellingPlanGroupsCreate and sellingPlanGroupsUpdate mutations
```

Comment:
Will running this for products or variants change how a dev would call these mutations? If so, should we have an example of how to tell which one we're rendering?